internal/bminventory/inventory.go

```diff
@@ -5412,8 +5412,25 @@ func (b *bareMetalInventory) DeregisterInfraEnvInternal(ctx context.Context, par
 	log := logutil.FromContext(ctx, b.log)
 	var infraEnv *common.InfraEnv
 	var err error
+	success := false
 	log.Infof("Deregister infraEnv id %s", params.InfraEnvID)
 
+	defer func() {
+		if success {
+			msg := fmt.Sprintf("Successfully deregistered InfraEnv %s", params.InfraEnvID.String())
+			log.Info(msg)
+			eventgen.SendDeregisteredInfraEnvEvent(ctx, b.eventsHandler, params.InfraEnvID)
+		} else {
+			errWrapperLog := log
+			if err != nil {
+				errWrapperLog = log.WithError(err)
+			}
+			errString := fmt.Sprintf("Failed to deregister InfraEnv %s", params.InfraEnvID.String())
+			errWrapperLog.Errorf(errString)
+			eventgen.SendInfraEnvDeregisterFailedEvent(ctx, b.eventsHandler, params.InfraEnvID, errString)
```

Comment:
here I'd use err.Error() instead of errString which contains the entire message.
have a look at the produced event and see the repetition, should be something like:
```
"Failed to deregister infra env. Error: Failed to deregister InfraEnv UUID"
```
while it would be useful to record only the error
```
"Failed to deregister infra env. Error: abc"  // as reported from err.Error() if err isn't nil.

Reply:
Thanks for the review.

Done.
I'm appending `err.Error()` to `errString` in case it is not `nil` and it prints the msg once.
Let me know if that captures your comment currectly.